addons/sale_purchase/models/product_template.py

```diff
@@ -9,23 +9,26 @@ class ProductTemplate(models.Model):
     _inherit = 'product.template'
 
     service_to_purchase = fields.Boolean(
-        "Subcontract Service",
-        compute='_compute_service_to_purchase', store=True, readonly=False,
+        "Subcontract Service", company_dependent=True,
         help="If ticked, each time you sell this product through a SO, a RfQ is automatically created to buy the product. Tip: don't forget to set a vendor on the product.")
 
-    _sql_constraints = [
-        ('service_to_purchase', "CHECK((type != 'service' AND service_to_purchase != true) or (type = 'service'))", 'Product that is not a service can not create RFQ.'),
-    ]
-
-    @api.constrains('service_to_purchase', 'seller_ids')
+    @api.constrains('service_to_purchase', 'seller_ids', 'type')
     def _check_service_to_purchase(self):
         for template in self:
-            if template.service_to_purchase and not template.seller_ids:
-                raise ValidationError(_(
-                    "Please define the vendor from whom you would like to purchase this service automatically."))
+            if template.service_to_purchase:
+                if template.type != 'service':
+                    raise ValidationError(_("Product that is not a service can not create RFQ."))
+                elif not template.seller_ids:
+                    raise ValidationError(_("Please define the vendor from whom you would like to purchase this service automatically."))
 
-    @api.depends('type', 'expense_policy')
-    def _compute_service_to_purchase(self):
-        for template in self:
-            if template.type != 'service' or template.expense_policy != 'no':
-                template.service_to_purchase = False
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get('service_to_purchase') and not vals.get('seller_ids'):
+                raise ValidationError(_("Please define the vendor from whom you would like to purchase this service automatically."))
```

Comment:
@IT-Ideas , At the time of record creation it will only add the stored field values in super create of model and for company_dependent and non-stored field values are not updated correctly while calling the constraint so the constraint is not fired while creating record due to invalid value in self. that's why we need to override and check the values in create method.
for ref. https://github.com/odoo/odoo/blob/master/odoo/models.py#L4183-L4217, Thanks